benchbuild/source/git.py

```diff
@@ -73,14 +72,14 @@ def version(self, target_dir: str, version: str = 'HEAD') -> pb.LocalPath:
         """
         src_loc = self.fetch()
         tgt_loc = local.path(target_dir) / self.local
-        lock_file = local.path(target_dir) / self.local + '.lock'
-
-        worktree = git['worktree']
-        with local.cwd(src_loc):
-            mkdir('-p', tgt_loc)
-            with flocked(lock_file):
-                worktree('prune')
-                worktree('add', '--detach', tgt_loc, version)
+        clone = git['clone']
+        checkout = git['checkout']
+
+        mkdir('-p', tgt_loc)
+        with local.cwd(tgt_loc):
+            clone('--dissociate', '--reference', src_loc, self.remote, '.')
```

Comment:
If I'm correct, this should just be a copy from tmp in the background or?

Reply:
It should have the same effect, yes. I'm not sure if dissociate will resolve the borrowed references eager or lazy.
Without a container-context, it would be possible to leave the dissociate option out and save significant disk space.